src/Juvix/Core/Erasure/Types.hs

```diff
@@ -1,32 +1,38 @@
 module Juvix.Core.Erasure.Types where
 
 import qualified Juvix.Core.Erased as Erased
+import qualified Juvix.Core.IR.Types as IR
 import Juvix.Library hiding (empty)
 
-data Env primTy
+data Env primTy primVal
   = Env
       { typeAssignment ∷ Erased.TypeAssignment primTy,
+        context ∷ IR.Context primTy primVal,
         nextName ∷ Int,
         nameStack ∷ [Int]
       }
   deriving (Show, Eq, Generic)
 
-newtype EnvErasure primTy a = EnvEra (ExceptT ErasureError (State (Env primTy)) a)
+newtype EnvErasure primTy primVal a = EnvEra (ExceptT ErasureError (State (Env primTy primVal)) a)
```

Comment:
we should probably eventually write a type alias for the main type here so we don't repeat it for every deriving via.

(Template haskell may be worth looking into for this)